### ISSUE TITLE
2.2.10 fail to buil on arm and i686

### ISSUE THREAD

**Issue by Erick555** [apple/cups#5483]
Hello again,

W noticed that cups 2.2.10 fails to build on arm and i686 arches (x86-64 and aarch64 work good) in linux with the following error:
```
hash.c:238:13: error: conversion to ‘ssize_t’ {aka ‘int’} from ‘unsigned int’ may change the sign of the result [-Werror=sign-conversion]
             return (gnutls_hash_get_len(alg));
                    ~^~~~~~~~~~~~~~~~~~~~~~~~~
```
Build config:
```
    CFLAGS: -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions
      -fstack-protector-strong -grecord-gcc-switches
    CXXFLAGS: -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions
      -fstack-protector-strong -grecord-gcc-switches
    HOME: /tmp
    LC_ALL: en_US.UTF-8
    LDFLAGS: -Wl,-z,relro,-z,now -Wl,--as-needed
    LOGNAME: tomjon
    LT_SYS_LIBRARY_PATH: /usr/lib/arm-linux-gnueabihf
    MAKEFLAGS: -j8
    PATH: /usr/bin:/bin:/usr/sbin:/sbin
    PYTHON: /usr/bin/python3
    PYTHONHASHSEED: '0'
    SHELL: /bin/sh
    SOURCE_DATE_EPOCH: '1320937200'
    TERM: dumb
    TZ: UTC
    V: '1'


    configure --prefix=/usr \
    --exec-prefix=/usr \
    --bindir=/usr/bin \
    --sbindir=/usr/bin \
    --sysconfdir=/etc \
    --datadir=/usr/share \
    --includedir=/usr/include \
    --libdir=/usr/lib/arm-linux-gnueabihf \
    --libexecdir=/usr/libexec \
    --localstatedir=/var \
    --sharedstatedir=/usr/com \
    --mandir=/usr/share/man \
    --infodir=/usr/share/info  --enable-deterministic-archives \
    --enable-shared \
    --disable-static \
    --build=arm-unknown-linux-gnueabihf \
    --host=arm-unknown-linux-gnueabihf --enable-debug \
    --with-components=core \
    --with-dbusdir=/etc/dbus-1 \
    --disable-systemd \
    --enable-threads \
    --enable-gnutls \
    localedir=/usr/share/locale \
    DSOFLAGS="$CFLAGS $LDFLAGS"
```
[Full build log from arm](https://freedesktop-sdk.gitlab.io/-/freedesktop-sdk/-/jobs/147225357/artifacts/cache/buildstream/logs/freedesktop-sdk/desktop-cups/1aff57a0-build.4378.log)

In [debian build](https://buildd.debian.org/status/fetch.php?pkg=cups&arch=armhf&ver=2.2.10-3&stamp=1544790986&raw=0) and [fedora build](https://kojipkgs.fedoraproject.org//packages/cups/2.2.10/1.fc30/data/logs/armv7hl/build.log) there seem to be the same warning but the build continues. Do you have any ideas?

**Comment by michaelrsweet** [comment]
Yes, GNU TLS needs to settle on return types for their APIs. The "fix" will be to add a cast to the affected call to gnutls_hash_get_len.

**Comment by Erick555** [comment]
The thing which breaks the build for us is that we build cups using git as source, not tarballs and cups treats them as devel ones and makes errors fatal: https://github.com/apple/cups/blob/7d730d0ce72263e1b53e97bce4e45d08be21e564/config-scripts/cups-compiler.m4#L184

Is it possible to introduce some `--devel` configure flag instead treating all builds from git as devel ones?

**Comment by Erick555** [comment]
Sure, it's better to address it however all linux distros simply ignored it because they don't build from git which means all the burden is for those who do 😄 .

**Comment by michaelrsweet** [comment]
[master 0057f146f] Fix compile error on Linux (Issue #5483)

[branch-2.2 328d86397] Fix compile error on Linux (Issue #5483)
